### PR TITLE
對imxrt的SPI傳輸改用非阻塞API

### PR DESCRIPTION
**Pull request by LeeChunHei**
## 拉取/合并请求描述：(PR description)

[
對imxrt的SPI傳輸改用非阻塞API
對imxrt的SPI傳輸使用信号量，確保傳輸完成后才退出傳輸函數
Use non-blocking api for spi transfer in imxrt driver
Use a semaphore to ensure xfer function return while the transfer is completed

已經在自製的imxrt1052 板子進行了測試
Had tested with a custom made imxrt1052 board
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/imxrt/libraries/drivers/drv_spi.c [hunk 1]
```diff
@@ -325,12 +347,13 @@ static rt_uint32_t spixfer(struct rt_spi_device *device, struct rt_spi_message *
 
     if(RT_FALSE == spi->dma_flag)
     {
-        status = LPSPI_MasterTransferBlocking(spi->base, &transfer);
+        status = LPSPI_MasterTransferNonBlocking(spi->base, &spi->spi_normal, &transfer);
```
Comment (xinyigao) [inline bsp/imxrt/libraries/drivers/drv_spi.c:350]: 添加spi的非阻塞模式和阻塞模式冲突，请添加可以提供：阻塞和非阻塞模式的选择

Comment (LeeChunHei) [reply bsp/imxrt/libraries/drivers/drv_spi.c:350]: 请问有什么冲突

**Comment by xinyigao** [comment]
在添加非阻塞模式支持的时候，最好确保也可以支持阻塞模式
发件人: LeeChunHei <notifications@github.com>
发送时间: 2021年2月22日 17:42
收件人: RT-Thread/rt-thread <rt-thread@noreply.github.com>
抄送: Xinyi Gao <xinyi.gao@nxp.com>; Comment <comment@noreply.github.com>
主题: [EXT] Re: [RT-Thread/rt-thread] imxrt的SPI鬏改用非阻塞API (#4336)


Caution: EXT Email

@LeeChunHei commented on this pull request.

________________________________

In bsp/imxrt/libraries/drivers/drv_spi.c<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FRT-Thread%2Frt-thread%2Fpull%2F4336%23discussion_r580106841&data=04%7C01%7Cxinyi.gao%40nxp.com%7C56cf72163285443a6ba208d8d71621af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637495837339598372%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=TCfaGef7fiAeyJm2w3AmVNaRk9s1jHZLbwZA%2BJYDfxw%3D&reserved=0>:

> @@ -325,12 +347,13 @@ static rt_uint32_t spixfer(struct rt_spi_device *device, struct rt_spi_message *







     if(RT_FALSE == spi->dma_flag)



     {



-        status = LPSPI_MasterTransferBlocking(spi->base, &transfer);



+        status = LPSPI_MasterTransferNonBlocking(spi->base, &spi->spi_normal, &transfer);



请问有什么冲突

―

File: bsp/imxrt/libraries/drivers/drv_spi.c (hunk 1 above)
Comment (LeeChunHei) [reply bsp/imxrt/libraries/drivers/drv_spi.c:353]: 添加了
透過定義BSP_USING_BLOCKING_SPI便可以使用阻塞模式

**Comment by LeeChunHei** [comment]
> 在添加非阻塞模式支持的时候，最好确保也可以支持阻塞模式 发件人: LeeChunHei <notifications@github.com> 发送时间: 2021年2月22日 17:42 收件人: RT-Thread/rt-thread <rt-thread@noreply.github.com> 抄送: Xinyi Gao <xinyi.gao@nxp.com>; Comment <comment@noreply.github.com> 主题: [EXT] Re: [RT-Thread/rt-thread] imxrt的SPI鬏改用非阻塞API (#4336) Caution: EXT Email @LeeChunHei commented on this pull request.
> ________________________________ In bsp/imxrt/libraries/drivers/drv_spi.c<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FRT-Thread%2Frt-thread%2Fpull%2F4336%23discussion_r580106841&data=04%7C01%7Cxinyi.gao%40nxp.com%7C56cf72163285443a6ba208d8d71621af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637495837339598372%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=TCfaGef7fiAeyJm2w3AmVNaRk9s1jHZLbwZA%2BJYDfxw%3D&reserved=0>:
> @@ -325,12 +347,13 @@ static rt_uint32_t spixfer(struct rt_spi_device *device, struct rt_spi_message *
> if(RT_FALSE == spi->dma_flag) { - status = LPSPI_MasterTransferBlocking(spi->base, &transfer); + status = LPSPI_MasterTransferNonBlocking(spi->base, &spi->spi_normal, &transfer); 请问有什么冲突 ― You are receiving this because you commented. Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FRT-Thread%2Frt-thread%2Fpull%2F4336%23discussion_r580106841&data=04%7C01%7Cxinyi.gao%40nxp.com%7C56cf72163285443a6ba208d8d71621af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637495837339598372%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=TCfaGef7fiAeyJm2w3AmVNaRk9s1jHZLbwZA%2BJYDfxw%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAJU2SMCCOZFAGQC747PRMT3TAIRHHANCNFSM4XS2MAAQ&data=04%7C01%7Cxinyi.gao%40nxp.com%7C56cf72163285443a6ba208d8d71621af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C637495837339608365%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=dn0Fz8OFbcqAEmCQ5gIreC9Vx7Ux8EIyjEAOnWjQOmM%3D&reserved=0>.

添加了
透過定義BSP_USING_BLOCKING_SPI便可以使用阻塞模式

**Review by xinyigao11 — APPROVED** [review]

**Pull request merged**